java-checks/src/main/java/org/sonar/java/checks/UselessImportCheck.java

```diff
@@ -223,13 +222,7 @@ public void visitTrivia(SyntaxTrivia syntaxTrivia) {
     }
 
     private void updatePendingImportsForComments(String comment) {
-      Iterator<String> it = pendingImports.iterator();
-      while (it.hasNext()) {
-        String pendingImport = it.next();
-        if (comment.contains(extractLastClassName(pendingImport))) {
-          it.remove();
-        }
-      }
+      pendingImports.removeIf(pendingImport -> comment.contains(extractLastClassName(pendingImport)));
```

Comment:
This is nice, could almost be a rule!

Reply:
made from intellij check but indeed @agigleux can you have a look ?